src/vs/base/node/id.ts

```diff
@@ -99,3 +100,16 @@ async function getMacMachineId(errorLogger: (error: any) => void): Promise<strin
 		return undefined;
 	}
 }
+
+const SQM_KEY: string = '\\Software\\Microsoft\\SQMClient';
+export async function getSqmMachineId(): Promise<string | undefined> {
```

Comment:
Would it be easier to pass an empty string around rather than an or type that is undefined. Compared to machineid which is always defined